src/rdkafka_msgset_writer.c

```diff
@@ -931,9 +936,18 @@ static int
 rd_kafka_msgset_writer_compress_lz4 (rd_kafka_msgset_writer_t *msetw,
                                      rd_slice_t *slice, struct iovec *ciov) {
         rd_kafka_resp_err_t err;
+        int compLevel =
+                msetw->msetw_rktp->rktp_rkt->rkt_conf.compression_level;
+        if (compLevel>RD_KAFKA_COMPLEVEL_LZ4_MAX) {
```

Comment:
Normalize these values in topic_new() instead so it doesn't need to be done for each messageset

Reply:
Right! done in f15ebaa